From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Scott Hess <shess@chromium.org>
Date: Thu, 12 Feb 2015 15:01:26 -0800
Subject: [PATCH 06/10] [fts3] Fix uninit variable in fts3EvalDeferredPhrase.

Chromium compiles with SQLITE_DISABLE_FTS4_DEFERRED since fts4 is
disabled.  That compiles away sqlite3Fts3DeferredTokenList() which
otherwise initialized pList and nList in fts3EvalDeferredPhrase().

BUG=458266
R=thakis@chromium.org
---
 third_party/sqlite/src/ext/fts3/fts3Int.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/sqlite/src/ext/fts3/fts3Int.h b/third_party/sqlite/src/ext/fts3/fts3Int.h
index 3c6501cea432..24c0f2940f33 100644
--- a/third_party/sqlite/src/ext/fts3/fts3Int.h
+++ b/third_party/sqlite/src/ext/fts3/fts3Int.h
@@ -487,7 +487,7 @@ int sqlite3Fts3DeferredTokenList(Fts3DeferredToken *, char **, int *);
 # define sqlite3Fts3DeferToken(x,y,z) SQLITE_OK
 # define sqlite3Fts3CacheDeferredDoclists(x) SQLITE_OK
 # define sqlite3Fts3FreeDeferredDoclists(x)
-# define sqlite3Fts3DeferredTokenList(x,y,z) SQLITE_OK
+# define sqlite3Fts3DeferredTokenList(x,y,z) ((*(y)=0),(*(z)=0),SQLITE_OK)
 #endif

 void sqlite3Fts3SegmentsClose(Fts3Table *);
--
2.18.0

